### DISCUSSION TITLE
is it possible to mix portrait and landscape in a pdf or docx format without diving into tex?

### DISCUSSION THREAD

**Discussion by donboyd5** [quarto-dev/quarto-cli#680] [Q&A] [answered]
Hi. Looking to have a pdf or docx that is mostly portrait, but that then shifts to landscape for a wide table and back to portrait afterward.

I see that it is possible to establish document-wide portrait or landscape via _quarto.yml code such as

```
---
format:
  pdf:
    documentclass: scrreprt
    classoption: portrait
---
```

But even in a book it doesn't look like I can override classoption from portrait to landscape for a given qmd file because, as I read the documentation, yaml in individual chapter qmd files is ignored (and when I tried anyway, it was ignored).

I can see that various people appear to have done this in Rmarkdown, for example discussed [here](https://github.com/rstudio/rmarkdown-cookbook/issues/16), and if I read it properly there are ways to do it by keeping tex output and manually editing the tex file afterward, but (a) I don't know a thing about tex, and (b) I prefer to avoid post-processing manual editing if possible.

With that as background, is there a relatively simple but perhaps as-yet undocumented way to mix portrait and landscape in pdf and docx output formats in quarto?

Also, as I read the documentation, it may be possible to divide a project into separate folders, with a separate _metadata.yml file in each folder, and perhaps that is a way to do it, but then I wonder, (a) can I set it up so that the separate documents are automatically combined (I haven't seen anything in the documentation on that yet), and (b) if so, will page numbers run consecutively and will other integrating actions such as a cohesive index be straightforward?

Many thanks.

**Comment by dragonstyle** [comment]
Right now we don't have a simple way (built in) to do this, though that would be a pretty sweet feature, especially if we could support it will cross format (though that will be very challenging of course). The solutions that you linked to (using the `pdflscape` package) seem like they should work fine for Quarto as well. For example, this does appear to work for me:

````

---
title: "Mixing portrait and landscape WITHOUT a header.tex file"
format: 
    pdf:
      header-includes: |
        \usepackage{pdflscape}
        \newcommand{\blandscape}{\begin{landscape}}
        \newcommand{\elandscape}{\end{landscape}}

---

Portrait
```{r}
summary(cars)
```

\newpage
\blandscape
Landscape
```{r}
summary(cars)
```
\elandscape

\newpage

```{r}
summary(cars)
```

````

I'm not sure on the Word side, but I think this is a good general direction for PDF output.

**Reply by donboyd5** [reply]
Thanks. This is super-helpful.

I see that the same approach works with book format and a separate preamble.tex file with the following content (adding package geometry, too, based on [this site](https://www.sascha-frank.com/landscape.html), but not sure yet if I need it):

```
\usepackage{pdflscape}
\usepackage{geometry}
\newcommand{\blandscape}{\begin{landscape}}
\newcommand{\elandscape}{\end{landscape}}
```

and referencing it in _quarto.yml as follows:

```
format:
  pdf:
    include-in-header: preamble.tex
    documentclass: scrreprt
    classoption: portrait
```

The one remaining (minor) problem, at least with pdf output (haven't tried docx yet) is that the page number is on the short edge of the page (left side when the page is running long from left-right, whereas I would like it to be on the bottom when the page is oriented that way). [This](https://www.sascha-frank.com/landscape.html) suggests a solution, but it sounds awkward - I'd prefer a fully integrated single-step solution.

if you have any advice on that, I'd appreciate it.

In any event, I'll take another look tomorrow. If I get things figured out, I'm happy to post a complete solution, if that is helpful to others.

Thanks again.

**Reply by dragonstyle** [reply]
I noticed that page number issue and wasn't sure whether that was 'good behavior' or not, though it seemed bad to me! This thread has a couple of other suggestions of approaches you could try (that basically look like what you've done but with different packages):

https://tex.stackexchange.com/questions/337/how-to-change-certain-pages-into-landscape-portrait-mode

FWIW If you'd like you should be able to also take anything in your `preamble.tex` file and just place it as a block in `_quarto.yml` under `header-includes` like:

```yaml
format:
  pdf:
    header-includes: |
      <contents of preamble.tex>
    documentclass: scrreprt
    classoption: portrait
```

Either should work and it should just be a matter of what works best for you (for example, `preamble.tex` could be easily generated or modified by an external script).

If you wouldn't mind sharing what you end up with, I think that would be great and the community would really benefit from your learning and work. Thank you!

**Comment by donboyd5** [comment]
Many thanks for these additional tips. I will investigate further and post
once I have something in a nice crisp (and fully working) form.

And thanks for the amazing quarto software. I always prefer to automate
everything, to minimize risks from manual work and to maximize
reproducibility. It seems perfect for that. Trying to learn the basics of
it (and tex, or latex I guess) as quickly as possible. Will try to post in
a day or two or three.

**Comment by donboyd5** [comment, marked as answer]
This is just an information update for now. Not ready to post a full solution to my specific problem as it will take a while to figure that out.

**Goal**
Generate pdf AND docx output with mixed portrait and landscape pages, and with option to have page numbering for landscape on bottom of long edge of page rather than on left side of short edge.

**pdf - status and issues**
The solution above (see "Mixing portrait and landscape WITHOUT a header.tex file") works fine for pdf with page number on left short edge of landscape page. A few notes:

- Using the tex (?) code that redefines the begin and end landscape commands (for example `\newcommand{\blandscape}{\begin{landscape}}`) is crucial. The reason for this is that otherwise hashtags will not be interpreted properly.  For example, if in the solution above, the text "Landscape" is changed to "# Landscape" the code above interprets this properly as _Heading 1_, but if instead of the `\blandscape` and `\elandscape` we simply used `\begin{landscape}` and `\end{landscape}` it would not be interpreted properly.
- The page number is on the left for the landscape page. If you look at a book with mixed portrait and landscape pages, this is how it is typically done, which I suppose makes sense as people can rotate a book manually as they are reading, so I can be ok with it. However, if you look at a lot of online documents with mixed orientations, where people do not rotate their computer screens or bodies as they are reading, it is NOT done this way. Instead, the page number is on the bottom of the long edge. Also, if you have a mixed orientation Word document and view it, the landscape page number is on the bottom, as it is if you save the docx file to pdf.
- It would be nice to have a way to put a landscape page number in a pdf on the bottom of the long edge.
- A lot of people have asked this question on the web (see [this](https://latex.org/forum/search.php?keywords=landscape+page+number&fid%5B0%5D=47), and [this](https://latex.org/forum/viewtopic.php?f=47&t=23336&p=79206&hilit=landscape#p79206), and [this](https://latex.org/forum/viewtopic.php?f=19&t=32467&p=109284&hilit=landscape+page+number#p109284), and [this](https://latex.org/forum/viewtopic.php?f=44&t=23141&p=78515&hilit=landscape#p78515)). However, most of the answers are of the form "that's a bad idea", which does not help here.
- One promising proposed latex solution I have found is [here](https://tex.stackexchange.com/questions/556939/koma-not-changing-page-sizes-in-the-mid-of-the-document), but I haven't had time to try to implement it.
- And I have neglected to test out the solutions shown [here](https://tex.stackexchange.com/questions/337/how-to-change-certain-pages-into-landscape-portrait-mode), linked to above. They look very promising. If one of those approaches works well, I'll post a full pdf solution with the page-number-at-bottom-of-landscape-page as a variant.

**docx - status and issues**
Unfortunately, pdf solutions do not work for docx, which I suppose makes sense because the pdf solution uses a package called pdflscape, and as far as I know so far, latex is not intended to convert to docx (perhaps there is a package that does that but I haven't really looked). And producing the output in pdf and converting to Word doesn't seem to work, at least with my converter (Adobe) in part because the page numbering is not created in a footer but rather in the body of the document.

- So it seems like I should be looking for a pandoc solution for docx (?).  
- Several people have tried to figure this out - see [this](https://groups.google.com/g/pandoc-discuss/c/bqUg2KjpQhY), for example.
- The one that appears most promising to me so far is [this](https://stackoverflow.com/questions/16173081/is-there-an-option-to-control-output-page-orientation-using-knitr-pander-pand), down at the bottom of the web page. But I haven't had time to investigate yet.

Will post solution if I find one.

**Comment by mcanouil** [comment]
Related:

- https://github.com/quarto-dev/quarto-cli/discussions/6104
- https://github.com/quarto-dev/quarto-cli/discussions/680
- https://github.com/quarto-dev/quarto-cli/issues/8512

**Comment by Fpadt** [comment]
I just installed the quarto pre-release 1.6.23 and landscape come out-of-the-box, see: [https://github.com/quarto-dev/quarto-cli/pull/10581](https://github.com/quarto-dev/quarto-cli/pull/10581),  
else you can use the extension from which it originates [edvinsyk/quarto-landscape](https://github.com/edvinsyk/quarto-landscape)]

**Reply by mcanouil** [reply]
See https://prerelease.quarto.org/docs/authoring/article-layout.html#landscape-mode

(upon release on https://quarto.org/docs/authoring/article-layout.html#landscape-mode)
